views/007_events.sql

```diff
@@ -195,4 +199,21 @@ $$ LANGUAGE plpgsql;
 CREATE OR REPLACE TRIGGER notification_update_enqueue
 AFTER INSERT OR UPDATE OR DELETE ON notifications
 FOR EACH ROW
-EXECUTE PROCEDURE notifications_trigger_function ();
\ No newline at end of file
+EXECUTE PROCEDURE notifications_trigger_function ();
+
+-- Handle component updates
+CREATE
+OR REPLACE FUNCTION handle_component_updates () RETURNS TRIGGER AS $$
+BEGIN
+  IF OLD.deleted_at IS NULL AND NEW.deleted_at IS NOT NULL THEN
+    DELETE FROM team_components WHERE component_id = OLD.id;
+  END IF;
```

Comment:
Can we also do deletions for component relationships and other related resources as well ?

Reply:
Also, I think these should be in a different file ? Like how we created a new one for canaries/check deletion handling

Doesn't belong in events.sql